tests/tests/browser/browser_nudges_homepage.js

```diff
@@ -0,0 +1,126 @@
+/* Any copyright is dedicated to the Public Domain.
+* http://creativecommons.org/publicdomain/zero/1.0/ */
+
+"use strict";
+
+// The path of the add-on file relative to `getTestFilePath`.
+const ADDON_PATH = "urlbar_nudges_experiment-1.0.0.zip";
+
+// Use SIGNEDSTATE_MISSING when testing an unsigned, in-development version of
+// the add-on and SIGNEDSTATE_PRIVILEGED when testing the production add-on.
+const EXPECTED_ADDON_SIGNED_STATE = AddonManager.SIGNEDSTATE_MISSING;
+// const EXPECTED_ADDON_SIGNED_STATE = AddonManager.SIGNEDSTATE_PRIVILEGED;
+
+add_task(async function init() {
+  let previousDefaultEngine = await Services.search.getDefault();
+  await initAddonTest(ADDON_PATH, EXPECTED_ADDON_SIGNED_STATE);
+
+  registerCleanupFunction(async function() {
+    Services.search.setDefault(previousDefaultEngine);
+  });
+});
+
+add_task(async function google() {
+  Services.search.setDefault(Services.search.getEngineByName("Google"));
+
+  await withAddon(async () => {
+    let testPromise = awaitAddonMessage("Page is engine homepage: true");
```

Comment:
So I'm not sure if `awaitAddonMessage` is the best way of testing here, but it works! Setting up these extension tests was a headache -- looking forward to seeing how you write your feature-specific tests to look for ways to improve mine

Reply:
There's one other example of unit testing. I wrote a query-categorization class for interventions: https://github.com/0c0w3/urlbar-search-interventions-experiment/blob/master/src/QueryScorer.js

It's a class in its own file. The test uses the head.js mini-framework/workflow described in the readme, so it requires building the add-on, copying it and the test files into a mozilla-central checkout, etc. It loads the add-on and then uses `Services.scriptloader.loadSubScript` to load the `QueryScorer` class from the add-on, and then unit-tests the class: https://github.com/0c0w3/urlbar-search-interventions-experiment/blob/master/tests/tests/browser/browser_QueryScorer.js#L134

I guess I could have simply copied QueryScorer.js into the tests directory and then declared it as a `support-file` in browser.ini, rather than loading it from the add-on.

Anyway, I mention this not because you should do this here but just to share knowledge/techniques.

Reply:
This is good to know, thank you. I didn't know about `Services.scriptloader.loadSubScript`.